py/kubeflow/testing/test_helper.py

```diff
@@ -53,12 +53,8 @@ def generate_xml(self):
     # want one underscore after junit.
     output_file = os.path.join(self.artifacts_dir,
                                "junit_" + self.name.replace("_", "-") + ".xml")
-    # junit_xml produces a list of test suites, but gubernator
-    # only parses a single test suite. So here we generate
-    # the xml using junit-xml and only output the first test
-    # suite in our output file
     xml_out = junit_xml.TestSuite.to_xml_string([self])
-    first_test_suite = ET.fromstring(xml_out)[0]
```

Comment:
Looking at the lens tests.
https://github.com/kubernetes/test-infra/blob/master/prow/spyglass/lenses/junit/lens_test.go#L303-L314
I think it should work with all the complex cases now.